### PR TITLE
New languages: OMG IDL, WebIDL and Slice.

### PR DESCRIPTION
**Pull request by jolkdarr**

### CODE REVIEW

**Comment by jolkdarr** [comment]
`AUTHORS.ru.txt` not updated.

File: src/languages/idl.js [hunk 1]
```diff
@@ -0,0 +1,58 @@
+/*
+Language: OMG IDL
```
Comment (marcoscaceres) [inline src/languages/idl.js]: Nit: regarding the name of the file, remember there are other IDLs, like web IDL. Maybe call this omgidl.js?

Comment (marcoscaceres) [inline src/languages/idl.js]: Alternatively, we could make sure this covers both IDLs... webIDL is based on OMG IDL, so totally possible.

Comment (jolkdarr) [inline src/languages/idl.js]: I think I should attempt the second option. However, some issues may arise from merging OMG IDL and WebIDL grammars. For instance, WebIDL keywords shouldn't be highlighted for a snippet using OMG IDL.

File: test/detect/webidl/default.txt [hunk 2]
```diff
@@ -0,0 +1,24 @@
+/// Main namespace.
+namespace Namespace {
+    interface interface_identifier { /* interface_members... */ };
```
Comment (marcoscaceres) [inline test/detect/webidl/default.txt]: Note that WebIDL namespaces are different to, say, C++ namespaces, in that they don't contain interfaces (only attributes or methods). 

Maybe replace the first namespace declaration with:

```
namespace VectorUtils {
  readonly attribute Vector unit;
  double dotProduct(Vector x, Vector y);
  Vector crossProduct(Vector x, Vector y);
};
```

But the other declarations are fine.

**Comment by marcoscaceres** [comment]
@jolkdarr, could you possibly split this into two separate commits, one for WebIDL and one for OMGIDL? I'd like to cherry-pick the WebIDL commit to start testing with in my own fork.

File: test/detect/webidl/default.txt (hunk 2 above)
Comment (jolkdarr) [inline test/detect/webidl/default.txt]: Fix in progress...

Comment (jolkdarr) [inline test/detect/webidl/default.txt]: What's the difference between a namespace and an interface? Both may include attributes and methods. Have they different bindings?

**Comment by jolkdarr** [comment]
New branch in progress: webidl
...

**Comment by marcoscaceres** [comment]
Wonder why those llvm tests are timing out on older node?

**Comment by yyyc514** [comment]
> @marcoscaceres I hope this is merged soon.

This was 2 years ago... but are these more "web standards" related and critical and therefore somehow going to bypass our "no new languages" rule... or is your comment simply just 2 years old and I should go ahead and explain the 3rd party language modules and see about getting a repo setup for this?

**Comment by marcoscaceres** [comment]
They are not critical, as far as I know.

**Comment by yyyc514** [comment]
> They are not critical, as far as I know.

Ok, assuming that's just an old comment then and proceeding with current policy. :-)

**Comment by yyyc514** [comment]
@jolkdarr 

**First of all, truly sorry for the long delay in anyone responding to this PR.**

Thank you so much for desiring to contribute and writing a language grammar.  Unfortunately, we currently are no longer accepting new languages into the core repository.  The existing maintainers simply don't have time to maintain new languages (fixing bugs, dealing with issues, etc).  **We'd still love to host your language as a 3rd party module though** - if you're willing to help maintain it.  

It's easy to turn your language into a module that anyone who likes can easily integrate with Highlight.js using a single line of code:

```html
<script src="/path/to/highlight.min.js"></script>
<!-- they'd simply add your language modules -->
<script src="/path/to/highlightjs-web-idl.js"></script> 
<script>hljs.initHighlightingOnLoad();</script>
```

It's also possible for users to rebuild the `highlight.js` library to include specific 3rd party modules, all in a single file (vs loading a bunch of 3rd party languages one at a time).

If you'd like to become the maintainer for this language then we're happy to create a new repository in the `highlightjs` organization and host the repository there.  That assumes you'd be open to reviewing issues and fixing bugs as they arise.  You're also welcome to host the repo yourself and we'll link to it from our README.

There isn't much in the way of how to docs on this yet, but there are sample repositories and it's honestly pretty easy.  We're still fleshing out an [official "spec" for 3rd party modules](https://github.com/highlightjs/highlight.js/issues/2328), but even after that's done it should require minimal changes from the examples we have already.  IE, if you wanted to get started now there isn't any big reason to wait.

My own `robots.txt` grammar repository is a great, simple example:

https://github.com/highlightjs/highlightjs-robots-txt

If you have any questions feel free to ask.

**Comment by yyyc514** [comment]
> OMG IDL, WebIDL and Slice

@jolkdarr I'm not sure how these 3 are related... usually I'd suggest 3 separate language repos but if they are strongly related (and going to share a single maintainer) then perhaps a single repo can work.  Let me know what you think and if you want to proceed with packaging them as 3rd party language modules.

The build support for multiple languages in a single repo isn't quite finished yet, but I've been a good sample case to test it out anyways - so perhaps this will be the first?

**Comment by jolkdarr** [comment]
@yyyc514 The three languages are just related because they allow to specify interfaces.
Of course, they may and should be provided with separate PRs.

**Comment by yyyc514** [comment]
@jolkdarr Want to make a pass at converting these into 3rd party language modules then?

**Comment by jolkdarr** [comment]
@yyyc514 OK to convert the PR.

Who shall create the following packages?
- https://github.com/highlightjs/highlightjs-omg-idl
- https://github.com/highlightjs/highlightjs-web-idl
- https://github.com/highlightjs/highlightjs-slice

**Comment by yyyc514** [comment]
I went with https://github.com/highlightjs/highlightjs-zeroc-slice.

I think we can still name the grammar "slice" though... sometimes I feel the repo names just need a little more context - since people really only need to use them once to install/use the grammar.

**Pull request closed without merging**